src/map/entities/mobentity.cpp

```diff
@@ -1275,6 +1275,12 @@ void CMobEntity::DropItems(CCharEntity* PChar)
 
     DropList_t* dropList = itemutils::GetDropList(m_DropID);
 
+    // if mob had no droplist but drops are added dynamically then create empty list
+    if (dropList == nullptr && PAI->EventHandler.hasListener("ITEM_DROPS"))
+    {
+        dropList = new DropList_t;
+    }
+
```

Comment:
From what I can recall, a mob dropID is 0 if they do not have any drop lists from the SQL file. In that case we should be given an empty `DropList_t` that it can use to pass to the `ITEM_DROPS` event. Can you go into detail about what is happening here?

Also this is a memory leak as we're creating a new `DropList_t` but it is never cleaned up.

Reply:
@jmcmorris The mobs dropID should be 0 if they do not have any items but in this case Critical did not change the dropID of Gration but left it as the old value which no longer had any rows, thus the dropList was null causing the issue. This code is a safety in case the dropID of such a mob is not 0 for some reason. Can remove it if we want to assume that all such mobs will have dropID of 0 in future.

Reply:
I meant to link this: https://github.com/AirSkyBoat/AirSkyBoat/blob/staging/src/map/utils/itemutils.cpp#L632

If the sql droplist is empty (we should probably go ahead and clean that up but even if it does linger it should still have a DropList_t created that is empty and consequently we don't need to create one here right? How are we ended up with a nullptr for the droplist?

Reply:
@jmcmorris Gration had dropID of 1218 in mob_groups but there was no row with dropID 1218 in mob_droplist

Reply:
Ahhh! That makes sense now. Honestly I think we should have an assert or something for that sort of thing. Another option is to fill `g_pDropList` with valid `DropList_t` ptrs so if this does happen it is the same as no dropID.

Reply:
But this is something you can handle earlier (during data loading) rather than every time a mob is killed and that is always a win.

Reply:
@jmcmorris I updated the PR with changes, let me know if you think they are reasonable